### ISSUE TITLE
Can't install xdebug-beta: Invalid version string "3.0.0beta1-beta"

### ISSUE THREAD

**Issue by mlocati** [FriendsOfPHP/pickle#191]
Installing xdebug-beta fails:

```sh
pickle install xdebug-beta

  - Installing xdebug (latest-beta): Downloading (100%)

In VersionParser.php line 176:

  Invalid version string "3.0.0beta1-beta"
```

Please remark that the Xdebug source code contains [`#define XDEBUG_VERSION    "3.0.0beta1"`](https://github.com/xdebug/xdebug/blob/3.0.0beta1/php_xdebug.h#L23)

**Comment by desrosj** [comment]
Using `pickle install xdebug-3.0.0beta1` is also failing. The result is `http error while loading information for xdebug-3.0.0beta1: unknown error`.

**Comment by pierrejoye** [comment]
hello,

you need to use Master, not the release to have the option to disable
server check. :)

**Comment by ericpoe** [comment]
I get a slightly different error message when building from the master branch: 'Invalid version string "3.0.0beta1-beta"'

Quick recreation in Docker:

```Dockerfile
FROM composer:1 AS composer-1
FROM php:7-fpm AS php7
COPY --from=composer-1 /usr/bin/composer /usr/bin/composer

RUN apt-get update && apt-get install -y \
  git \
  libzip-dev \
  && docker-php-ext-install -j$(nproc) zip

RUN mv "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"

RUN curl -LSs https://box-project.github.io/box2/installer.php | php \
  && git clone https://github.com/FriendsOfPHP/pickle.git \
  && cd pickle \
  && composer install --no-dev --optimize-autoloader \
  && php -d phar.readonly=0 box.phar build \
  && mv pickle.phar /usr/local/bin/pickle \
  && chmod +x /usr/local/bin/pickle

RUN pickle install xdebug-beta
```

**Comment by neclimdul** [comment]
Not 100% but this sounds like a dupe of #188. The fix for that hasn't landed in a release yet.

**Comment by bartvanhoutte** [comment]
Any workaround?

**Comment by hrs-o** [comment]
As of 2020.11.17, when I try to add xdebug-beta, I get the following error:
> Invalid version string "3.0.0RC1-beta"

The version definition of Xdebug itself seems to be `3.0.0RC1`, but it seems that https://github.com/FriendsOfPHP/pickle/blob/master/src/Package/Util/Dumper.php#L59 has added` -beta`.


I get an error when "3.0.0RC1-beta" is passed to https://github.com/composer/semver/blob/main/src/VersionParser.php#L102 .

Is this behavior correct?

**Comment by mlocati** [comment]
Xdebug 3.0.0 has just been released, but it's not possible to install it with the last published pickle version (0.6.0).
What about publishing a new pickle version (which includes #185)?
Building pickle by cloning the `master` branch requires composer and a bunch of PHP extensions...

**Comment by pierrejoye** [comment]
pls use the current git

**Comment by mlocati** [comment]
Using the current git requires composer, which in turn requires the openssl, (iconv || mbstring), zlib PHP extensions... what's wrong in publishing a new pickle version?

**Comment by pierrejoye** [comment]
Nothing wrong but time :)

**Comment by mlocati** [comment]
To speedup the release process, I can submit a pull request with a GitHub Action, that will be triggered by a tag creation. That action would do the following:

- builds the changelog by listing the commit messages since the previous tag
- creates the `pickle.phar` file
- create a draft release by using the collected changedlog and attaches the `pickle.phar` file

So, you'll only have to:
1. create a tag
2. wait for the GitHub Action to do the dirty job
3. publish the draft released created by the GitHub Action (eventually editing the release notes if needed)

**Comment by mlocati** [comment]
[Here's](https://github.com/mlocati/picke-sample/blob/master/.github/workflows/create-release.yml) what I meant.
With that action in place, I simply pushed a new tag (`v9.9.9`), and the action [created](https://github.com/mlocati/picke-sample/runs/1480251515) this draft:

![immagine](https://user-images.githubusercontent.com/928116/100748644-dccdce00-33e3-11eb-935e-6edfbb269e15.png)

**Comment by pierrejoye** [comment]
Sorry I missed that one.

That would be extremely cool yes.

:)

**Comment by mlocati** [comment]
> That would be extremely cool yes.

@pierrejoye See https://github.com/FriendsOfPHP/pickle/pull/205
Could we have a new 0.7.0 (or whatever) pickle version?

**Comment by mlocati** [comment]
This issue should have been fixed by #201
